packages/components/src/composite/stories/legacy.story.tsx

```diff
@@ -0,0 +1,244 @@
+/**
+ * External dependencies
+ */
+import type { Meta, StoryFn, StoryContext } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { isRTL } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	Composite,
+	CompositeItem,
+	CompositeGroup,
+	useCompositeState,
+} from '../legacy';
+
+import type { InitialState } from '../legacy';
+
+const Placeholder = ( _: InitialState ) => <></>;
+
+const transform = ( code: string, context: StoryContext ) => {
+	const config = ` ${ JSON.stringify( context.args, null, 2 ) } `;
+	const state = config.replace( ' {} ', '' );
+	return [
+		`const state = useCompositeState(${ state });`,
+		'',
+		'return (',
+		'  ' +
+			code
+				.replaceAll( /state=\{\{[\s\S]*?\}\}/g, 'state={ state }' )
+				.replaceAll( '=>', '' )
+				.replaceAll( /baseId=[^>]+?(\s*>)/g, ( _, close ) => {
+					return `{ ...state }${ close }`;
+				} )
+				.replaceAll( /<Composite\w+[\s\S]*?>/g, ( match ) =>
+					match.replaceAll( /\s+\s/g, ' ' )
+				)
+				.replaceAll(
+					/ >\s+([\w\s]*?)\s+<\//g,
+					( _, value ) => `>${ value }</`
+				)
+				.replaceAll( '} >', '}>' )
+				.replaceAll( /\n/g, '\n  ' ),
+		');',
+	].join( '\n' );
+};
+
+const meta: Meta< typeof Placeholder > = {
+	title: 'Components/Composite/Composite (Legacy)',
+	id: 'components-composite-legacy',
+	component: Placeholder,
+	args: {
+		rtl: isRTL() || undefined,
+	},
+	argTypes: {
```

Comment:
We shouldn't need to manually specify the `argTypes` — Storybook should extract this information from the props' TypeScript + JSDocs info

Reply:
The appropriate arg types to be rendered need to be derived from `useCompositeState`, rather than any of the consuming components. Specifying `useCompositeState` as the value for the `component` meta prop technically works, insomuch as the right arg types are generated, but Typescript doesn't like it because `useCompositeState` isn't a component. I tried with a `Placeholder` component that expected the same props, but that didn't work at all.

Reply:
I understand what's been done with the `Placeholder` component, although I find it a bit confusing that the stories on the `Composite` component actually show, as the available `props`, the arguments of the `useCompositeState` and `useCompositeStore` hooks.

I feel like we may want to display the `Composite` component props, if anything for coherency with the rest of the storybook example. What do you think, @mirka ?

We should also add all of the related `Composite*` components as `subcomponents` (see [example](https://github.com/WordPress/gutenberg/blob/ae7467d8bab6cbe538ac0de648b57a62480e2bdc/packages/components/src/dropdown-menu-v2/stories/index.story.tsx#L37-L54))

If we agree on this approach, it would apply to all 3 storybook files in the PR.

Reply:
Hmm, this is a difficult one.

- Unstable (Reakit): I think these stories were only for internal testing, and we should ultimately keep this out of the public docs.
- Legacy (Adapter): I guess we can justify showing the `useCompositeState` props here, because the Composite component actually does take those props. Though, we need to make it very clear that you're not supposed to pass individual props directly to the component.
- Current (Ariakit): I agree it'd be a bit confusing to show the `useCompositeStore` props here. What do we think about sending people to see the [Ariakit docs](https://ariakit.org/reference/use-composite-store) directly? It is a bit inconvenient that we can't fiddle with the props in a live playground, but this is a pretty advanced component so it's probably ok.

Reply:
I've added the _legacy_ subcomponents, and made it clear that the initial props are for `useCompositeState`. Additionally, after some reconfiguration of the `Placeholder` component, these properties are now derived directly from the code, rather than being hardcoded in `argTypes`. Unfortunately the same cannot be said of the `Composite*` components, which still claim that "interactive controls couldn't be auto-generated".

![Legacy composite docs, showing tabs for 'useCompositeState', 'Composite', 'CompositeGroup' and 'CompositeItem'](https://github.com/WordPress/gutenberg/assets/159848/3fc4a59b-69e9-4232-9fd2-a29849969dca)

Similarly, the _current_ docs now include a tab for `useCompositeStore` as well as for each `Composite*` component, each of which points to the relevant Ariakit documentation. I couldn't (quickly) see a way to override the table entirely, or to render an `EmptyTable` view, so this seemed to be the least-effort solution.

![Legacy composite docs, showing tabs for 'useCompositeStore', 'Composite', 'CompositeGroup', 'CompositeRow' and 'CompositeItem'](https://github.com/WordPress/gutenberg/assets/159848/d916b33f-8321-422c-ae49-9de5078632a3)

Happy to delete the unstable docs – the plan had always been to get rid of them as soon as Reakit is dropped anyway.